cornflow/endpoints/execution.py

```diff
@@ -159,11 +165,24 @@ def get(self, idx):
         and an integer with the HTTP status code.
         :rtype: Tuple(dict, integer)
         """
-        self.user_id, self.admin, self.super_admin = Auth.return_user_info(request)
-        status = ExecutionModel.get_one_execution_from_user(self.user_id, idx).finished
-        # TODO: call airflow to check status
-        if not status:
-            # Here we should call airflow to check solving status
-            pass
+        airflow_conf = dict(url=current_app.config['AIRFLOW_URL'],
+                            user=current_app.config['AIRFLOW_USER'],
+                            pwd=current_app.config['AIRFLOW_PWD'])
 
-        return {'finished': status}, 200
+        self.user_id, self.admin, self.super_admin = Auth.return_user_info(request)
+        execution = ExecutionModel.get_one_execution_from_user(self.user_id, idx)
+        status = execution.finished
+        if status:
+            return {'status': 'finished'}, 200
+        dag_run_id = execution.dag_run_id
```

Comment:
This is what I was refering on the other comment, but come to think of it if we want to change this column, we should keep the following in mind:

- We should have a column with a meaningful way to show the status of the run: running, failed, finished correctly. These are the three states that I can think as of right now.
- We should have an extra column, that can have null values, with a meaningful description of what happened (mainly to store failed messages).

Does this make sense? or should be simplified? Because if for example the error is that Airflow is not available, that is not going to be stored on the log_json or log_text fields.

Reply:
I agree with the two column approach. The "failed", "running" and "finished" could be three integer codes: -1, 0 and 1 respectively. And then we have the additional column with the error message. It reminds me of aimms pro.

Reply:
I agree with the two column approach. The "failed", "running" and "finished" could be three integer codes: -1, 0 and 1 respectively. And then we have the additional column with the error message. It reminds me of aimms pro.

Reply:
Ok, i will do this change in a new branch. Then we have to modify the way the status is checked and do:

```
status = execution.state == 1
if status:
    ...
```

or: 
```
if execution.state == 1:
    ...
```

Because i guess you want -1 for failure, 0 for running and 1 for finished